Source/Scene/GetFeatureInfoFormat.js

```diff
@@ -0,0 +1,283 @@
+     *        MIME type such as 'application/json' or text/xml'.  If this parameter is not specified, the provider will request 'json'
+     *        using 'application/json', 'xml' using 'text/xml', 'html' using 'text/html', and 'text' using 'text/plain'.
+     * @param {Function} [options.callback] A function to invoke with the GetFeatureInfo response from the WMS server
+     *        in order to produce an array of picked {@link ImageryLayerFeatureInfo} instances.  If this parameter is not specified,
+     *        a default function for the type of response is used.
+     */
+    var GetFeatureInfoFormat = function(type, format, callback) {
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(type)) {
+            throw new DeveloperError('type is required.');
+        }
+        //>>includeEnd('debug');
+
+        this.type = type;
+
+        if (!defined(format)) {
+            if (type === 'json') {
+                format = 'application/json';
+            } else if (type === 'xml') {
+                format = 'text/xml';
+            } else if (type === 'html') {
+                format = 'text/html';
+            } else if (type === 'text') {
+                format = 'text/plain';
+            }
+            //>>includeStart('debug', pragmas.debug);
+            else {
+                throw new DeveloperError('format is required when type is not "json", "xml", "html", or "text".');
+            }
+            //>>includeEnd('debug');
+        }
+
+        this.format = format;
+
+        if (!defined(callback)) {
+            if (type === 'json') {
+                callback = geoJsonToFeatureInfo;
+            } else if (type === 'xml') {
+                callback = xmlToFeatureInfo;
+            } else if (type === 'html') {
+                callback = textToFeatureInfo;
+            } else if (type === 'text') {
+                callback = textToFeatureInfo;
+            }
+            //>>includeStart('debug', pragmas.debug);
+            else {
+                throw new DeveloperError('callback is required when type is not "json", "xml", "html", or "text".');
+            }
+            //>>includeEnd('debug');
+        }
+
+        this.callback = callback;
+    };
+
+    function geoJsonToFeatureInfo(json) {
+        var result = [];
+
+        var features = json.features;
+        for (var i = 0; i < features.length; ++i) {
+            var feature = features[i];
+
+            var featureInfo = new ImageryLayerFeatureInfo();
+            featureInfo.data = feature;
+            featureInfo.properties = feature.properties;
+            featureInfo.configureNameFromProperties(feature.properties);
+            featureInfo.configureDescriptionFromProperties(feature.properties);
+
+            // If this is a point feature, use the coordinates of the point.
+            if (feature.geometry && feature.geometry.type === 'Point') {
```

Comment:
Not a good idea with JSON, because JSON can represent a null but can't represent undefined.  At best I could change this to `defined(feature.geometry) && feature.geometry !== null` but I think what I have is better.